Consul.Test/OperatorTest.cs

```diff
@@ -98,11 +101,26 @@ public async Task Segment_List()
         }
 
         [EnterpriseOnlyFact]
-        public async Task Operator_CreateArea()
+        public async Task Operator_AreaCreate()
         {
-            var check = new AreaRequest { PeerDatacenter = "dc2", UseTLS = false, RetryJoin = null };
-            var response = await _client.Operator.CreateArea(check);
+            var peerDataCenter = KVTest.GenerateTestKeyName();
+            var check = new AreaRequest { PeerDatacenter = peerDataCenter, UseTLS = false, RetryJoin = null };
+
+            var response = await _client.Operator.AreaCreate(check);
             Assert.NotNull(response.Response);
         }
+        [EnterpriseOnlyFact]
+        public async Task Operator_AreaList()
+        {
+            var peerDataCenter = KVTest.GenerateTestKeyName();
+            var area = new AreaRequest { PeerDatacenter = peerDataCenter, UseTLS = false, RetryJoin = null };
+
+            await _client.Operator.AreaCreate(area);
+
+            var req = await _client.Operator.AreaList();
+            Assert.NotEmpty(req.Response);
+            Assert.Equal(area.PeerDatacenter, req.Response.First().PeerDatacenter);
```

Comment:
There can be multiple items in the result list, so the first one is not necessarily the one we are looking for. You need to look for the right one, e.g.: 
```
var result = req.Response.Single(x => x.PeerDatacenter == area.peerDataCenter);
Assert.Equal(area.UseTLS, result.UseTLS);
Assert.Equal(area.RetryJoin, result.RetryJoin);  
```

Reply:
@marcin-krystianc I'm not sure if testing RetryJoin is sufficient with this approach ```Assert.Equal(area.RetryJoin, result.RetryJoin);``` 
or if it would be better to test each element individually in the array, like this: ```for (int i = 0; i < retryJoinAddresses.Length; i++)
    {
        Assert.Equal(retryJoinAddresses[i], createdArea.RetryJoin[i]);
    }```

Reply:
> @marcin-krystianc I'm not sure if testing RetryJoin is sufficient with this approach `Assert.Equal(area.RetryJoin, result.RetryJoin);` or if it would be better to test each element individually in the array, like this: `for (int i = 0; i < retryJoinAddresses.Length; i++) { Assert.Equal(retryJoinAddresses[i], createdArea.RetryJoin[i]); }`

There is an overload for `IEnumerable`:
```
    public static void Equal<T>(
      IEnumerable<T> expected,
      IEnumerable<T> actual,
      IEqualityComparer<T> comparer)
```

so I guess it will do the right thing. 
You can try to define manually two lists  and see how it reacts to different item order in both lists.

Reply:
![image](https://github.com/user-attachments/assets/b6f6dc4e-bd45-4277-b8c8-b829367d515f)
Tests 1 and 3 are passing, but test 2 is failing so its ok. I have pushed the changes please let me know if there's anything else to change.